File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/service/DbService.java
package org.javamaster.invocationlab.admin.service;

import com.alibaba.druid.DbType;
import org.apache.commons.lang3.tuple.Triple;
import org.javamaster.invocationlab.admin.model.erd.Column;
import org.javamaster.invocationlab.admin.model.erd.CommonErdVo;
import org.javamaster.invocationlab.admin.model.erd.DbsBean;
import org.javamaster.invocationlab.admin.model.erd.ErdOnlineModel;
import org.javamaster.invocationlab.admin.model.erd.ModulesBean;
import org.javamaster.invocationlab.admin.model.erd.SqlExecResVo;
import org.javamaster.invocationlab.admin.model.erd.Table;
import org.javamaster.invocationlab.admin.model.erd.TokenVo;
import org.springframework.http.MediaType;

import java.util.List;

public interface DbService {
    void checkDb(DbsBean dbsBean);

    List<String> getDbNames(DbsBean dbsBean);

    List<Table> getTables(DbsBean dbsBean, String selectDB);

    DbType getDbType();

    List<SqlExecResVo> execUpdate(DbsBean dbsBean, TokenVo tokenVo, CommonErdVo reqVo);

    List<Column> getTableColumns(DbsBean dbsBean, String selectDB, String tableName);

    SqlExecResVo execSql(CommonErdVo reqVo, DbsBean dbsBean, TokenVo tokenVo);

    Triple<String, MediaType, byte[]> exportSql(DbsBean dbsBean, TokenVo tokenVo, CommonErdVo reqVo);

    Integer getTableRecordTotal(CommonErdVo reqVo, DbsBean dbsBean, TokenVo tokenVo) throws Exception;

    ModulesBean refreshModule(ErdOnlineModel erdOnlineModel, String moduleName);

}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/service/DbService.kt
package org.javamaster.invocationlab.admin.service

import org.javamaster.invocationlab.admin.config.ErdException
import org.javamaster.invocationlab.admin.model.erd.*
import org.javamaster.invocationlab.admin.service.impl.MySqlDbServiceImpl
import org.javamaster.invocationlab.admin.service.impl.PostgreSqlDbServiceImpl
import com.alibaba.druid.DbType
import org.apache.commons.lang3.tuple.Triple
import org.springframework.http.MediaType

interface DbService {
    fun checkDb(dbsBean: DbsBean)

    fun getCheckSql(): String

    fun getDbNames(dbsBean: DbsBean): List<String>

    fun getTables(dbsBean: DbsBean, selectDB: String): List<Table>

    fun getDbType(): DbType

    fun execUpdate(dbsBean: DbsBean, tokenVo: TokenVo, reqVo: CommonErdVo): List<SqlExecResVo>

    fun getTableColumns(dbsBean: DbsBean, selectDB: String, tableName: String): List<Column>

    fun execSql(reqVo: CommonErdVo, dbsBean: DbsBean, tokenVo: TokenVo): SqlExecResVo

    fun exportSql(dbsBean: DbsBean, tokenVo: TokenVo, reqVo: CommonErdVo): Triple<String, MediaType, ByteArray>

    fun getTableRecordTotal(reqVo: CommonErdVo, dbsBean: DbsBean, tokenVo: TokenVo): Int

    fun refreshModule(erdOnlineModel: ErdOnlineModel, moduleName: String): ModulesBean

    companion object {
        fun getInstance(select: String): DbService {
            val dbService = DB_MAP[select] ?: throw ErdException("暂时不支持当前数据库:$select")
            return dbService
        }

        private val DB_MAP: Map<String, DbService> = object : HashMap<String, DbService>() {
            init {
                put("MYSQL", MySqlDbServiceImpl())
                put("PostgreSQL", PostgreSqlDbServiceImpl())
            }
        }
    }
}
